lib/modules.nix

```
@@ -510,7 +511,7 @@ let
       # context on the explicit arguments of "args" too. This update
       # operator is used to make the "args@{ ... }: with args.lib;" notation
       # works.
-    in f (args // extraArgs);
+    in f (builtins.intersectAttrs (functionArgs f) (args // extraArgs));
```

Looks like performance can improve slightly with this change:

```suggestion
    in f extraArgs;
```

(and a rename would be in order)